### PR TITLE
PrathibaJee/issue256

### PR DESCRIPTION
**Pull request by PrathibaJee**

### CODE REVIEW

**Review by openBackhaul — APPROVED** [review]

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- to configure and automate the forwardings between applications (please refer chapter about ForwardingList to learn more about forwarding)
+- to configure the profile instances in the ProfileCollection
+- to log the service request to the Execution And Trace Log(EATL) and OAM Log Application.
+- to authenticate the OAM request in AdministratorAdministration Application.
+
+    ![ApplicationPatternPackages](Images/ApplicationPatternPackages.png)
+
+## Application Pattern Modules
+
+ApplicationPattern modules consists of the followin sub-modules
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md:17]: please check the spelling of **following**

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfAuthenticationAuthorization/ConceptOfAuthenticationAuthorization.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+In all the approved applications of ApplicationPattern , 
+* OAM services are authorized by using BasicAuth 
+* Both the Individual and Basic services are authorized by using an operation-key
+
+### OAM service :
+
+ ![BasicAuth](Images/BasicAuth.png)
+* Every OAM service request is authorized by using BasicAuth. 
+* Usually, an administrator shall access the OAM services by using a username and password. 
+* When an application receives an OAM request , the BasicAuth will be validated with the help of one of the Tiny Application Controller called “AdministratorAdministration”(AA).
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfAuthenticationAuthorization/ConceptOfAuthenticationAuthorization.md:11]: please change it as "with the help of one of the applications in Tiny Application Controller"

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md (hunk 1 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md:17]: Thank you. Changed "followin" to "following"

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfAuthenticationAuthorization/ConceptOfAuthenticationAuthorization.md (hunk 2 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfAuthenticationAuthorization/ConceptOfAuthenticationAuthorization.md:11]: Thank you. Changed the statement as proposed.

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Individual services
+
+ * All the individual request are specific the current application's use-case.
+ * To understand the requirement , the Open API Specification and load file shared for this application needs to be understood.
+ * All the Individual request needs to send the following three parameters in the response header , 
+   * x-correlator : s'UUID for the service execution flow that allows to correlate requests and responses. Its value must be identical at the response compared with its corresponding request'
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md:6]: please remove "s'"

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+   * exec-time : reporting the total elapsed time for the execution, including all the additional processing needed to retrieve the data from the backend service. Expressed in milliseconds'
+   * life-cycle-state : Life cycle state of the consumed service
+   * backend-time : reporting the elapsed time for data retrieval from the backend (service invocation, database access…). Expressed in milliseconds
+ * Using the JSONDriver in the DatabaseDriver module of the ApplicationPattern the manipulation of the CONFIGFile this shall be done with minimum effort.
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md:10]: please check the sentence formation and modify accordingly

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+   * x-correlator : s'UUID for the service execution flow that allows to correlate requests and responses. Its value must be identical at the response compared with its corresponding request'
+   * exec-time : reporting the total elapsed time for the execution, including all the additional processing needed to retrieve the data from the backend service. Expressed in milliseconds'
+   * life-cycle-state : Life cycle state of the consumed service
+   * backend-time : reporting the elapsed time for data retrieval from the backend (service invocation, database access…). Expressed in milliseconds
+ * Using the JSONDriver in the DatabaseDriver module of the ApplicationPattern the manipulation of the CONFIGFile this shall be done with minimum effort. 
+ * All the Individual requests shall be approved by AdministratorAdministration application. Please refer [Authorization](../ConceptOfAuthenticationAuthorization/ConceptOfAuthenticationAuthorization.md) for further details.
+ * All the individual requests shall be logged to the OAMLog application. Please refer [logging](../ConceptOfLogDirection/ConceptOfLogDirection.md) section for further details.
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md:12]: please modify individual requests are logged to "ExecutionAndTraceLog(EaTL)" application

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+ * All the Individual request needs to send the following three parameters in the response header , 
+   * x-correlator : s'UUID for the service execution flow that allows to correlate requests and responses. Its value must be identical at the response compared with its corresponding request'
+   * exec-time : reporting the total elapsed time for the execution, including all the additional processing needed to retrieve the data from the backend service. Expressed in milliseconds'
+   * life-cycle-state : Life cycle state of the consumed service
+   * backend-time : reporting the elapsed time for data retrieval from the backend (service invocation, database access…). Expressed in milliseconds
+ * Using the JSONDriver in the DatabaseDriver module of the ApplicationPattern the manipulation of the CONFIGFile this shall be done with minimum effort. 
+ * All the Individual requests shall be approved by AdministratorAdministration application. Please refer [Authorization](../ConceptOfAuthenticationAuthorization/ConceptOfAuthenticationAuthorization.md) for further details.
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md:11]: please modify Individual services are authorized by operation-key

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfLogDirection/ConceptOfLogDirection.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+- timestamp : Date and time when response was sent
+
+The logged information shall be useful to track the configuration changes. 
+OL provided the following services to track the configuration changes , 
+* list-records
+* list-records-of-application
+
+### Basic and Individual Service Request
+
+ ![EATLFlow](Images/EATLFlow.png)
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfLogDirection/ConceptOfLogDirection.md:27]: EATLFlow.png should be changed

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfOAMServices/ConceptOfOAMServices.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# OAM services
+
+ * OaM layer makes the internal resources of the application available for configuration.
+ * OAM services are used to view and edit the entities of the CONFIGFile.
+ * Using the JSONDriver in the DatabaseDriver module of the ApplicationPattern the manipulation of the CONFIGFile this shall be done with minimum effort.
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ImplementingApplication/ConceptOfOAMServices/ConceptOfOAMServices.md]: please modify the line

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfLogDirection/ConceptOfLogDirection.md (hunk 7 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfLogDirection/ConceptOfLogDirection.md]: Many thanks. Updated the picture of EATLFlow.png

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md (hunk 3 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md]: Thanks. Removing the s'.

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md (hunk 4 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md]: Thanks. Modified the line.

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md (hunk 6 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md]: Thanks. Modified from OAMLog(OL) to ExecutionAndTraceLog(EaTL)

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md (hunk 5 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfIndividualServices/ConceptOfIndividualServices.md]: Thanks.Modified from OAMLog(OL) to ExecutionAndTraceLog(EaTL)

File: doc/ImplementingApplications/ImplementingApplication/ConceptOfOAMServices/ConceptOfOAMServices.md (hunk 8 above)
Comment (PrathibaJee) [reply doc/ImplementingApplications/ImplementingApplication/ConceptOfOAMServices/ConceptOfOAMServices.md]: Modified the line as proposed

**Review by vanithavalluripalli9 — APPROVED** [review]
Approving the request as mentioned review comments are resolved

**Review by vanithavalluripalli9 — APPROVED** [review]

**Review by openBackhaul — APPROVED** [review]

**Pull request merged**